assets/javascripts/lib/blockstack_client.js

```diff
@@ -8,3 +8,23 @@ function getUserSession(scopes) {
 
 export const privateUserSession = getUserSession(['store_write', 'publish_data']);
 export const publicUserSession = getUserSession([]);
+
+const options = {
+  redirectTo: '/',
+  finished: ({ }) => {
+    window.location = Constants.BLOCKSTACK_REDIRECT_URI;
+  },
+  userSession: privateUserSession,
+  appDetails: {
+    name: 'Envelop',
+    icon: 'https://envelop.app/images/manifest-icon.png',
+  },
+};
+
+const extensionOptions = options;
```

Comment:
@joaodiogocosta my JavaScript knowledge is limited, does this "clone" the object or is it a simple "reference"?
If this behaves like a reference that would explain the issue about the redirect to the extension...

Reply:
Yep, "reference":

![image](https://user-images.githubusercontent.com/1585289/95069815-227d5b80-06ff-11eb-9308-c4eb13caf6ae.png)

To shallow copy an object you can use:

```js
const optionsA = { ... };
const optionsB = Object.assign({}; optionsA}; // ES5
const optionsB = { ...optionsA }; // ES6
```

To deep copy you will need a lib like [lodash](https://lodash.com/docs/#cloneDeep):
```js
const options A = { ... };
const optionsB = _.cloneDeep(optionsA);
```

Reply:
I did the shallow copy (ES5, not sure if I should have done the ES6...), let me know :)